ДЗ по практической №1

arkadiyashikhmin opened the pull request.

qaashih committed:
Коммит с ДЗ по практической №1

AleksandrIlinskii commented on src/main/java/AddProduct.java:
```diff
@@ -0,0 +1,22 @@
+            if (answer.equalsIgnoreCase(("завершить"))) {
+                break;
+            }
+            else {
```
Else пишется на той же строке

AleksandrIlinskii commented on src/main/java/AddProduct.java:
```diff
@@ -0,0 +1,22 @@
+    static Scanner scanner = new Scanner(System.in);
+
+    public static void question() {
+        gameover = "Завершить";
```
Не стоит создавать static String, если ты его 1 раз присваиваешь и больше не меняешь
<img width="199" alt="image" src="https://github.com/arkadiyashikhmin/Java-Module-Project-YP/assets/129396632/1b523d07-ecb7-4de1-9ee6-d5773eef2b1b">

AleksandrIlinskii commented on src/main/java/AddProduct.java:
```diff
@@ -0,0 +1,22 @@
+    public static void question() {
+        gameover = "Завершить";
+        System.out.println("Хотите добавить еще один товар? Если нет, напишите \"завершить\".");
+        answer = scanner.nextLine();
```
Тут тоже, как и выше

AleksandrIlinskii commented on src/main/java/AddProduct.java:
```diff
@@ -0,0 +1,22 @@
+        System.out.println("Хотите добавить еще один товар? Если нет, напишите \"завершить\".");
+        answer = scanner.nextLine();
+        while (true) {
+            if (answer.equalsIgnoreCase(("завершить"))) {
```
У тебя уже есть переменная, которая содержит это слово, используй ее
<img width="285" alt="image" src="https://github.com/arkadiyashikhmin/Java-Module-Project-YP/assets/129396632/d15ec8fa-1f72-473d-9b42-659d9574b8f8">

AleksandrIlinskii commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,43 @@
+import java.util.Scanner;
 
 public class Main {
+    public static double sum;
```
Переменная нигде не используется
<img width="463" alt="image" src="https://github.com/arkadiyashikhmin/Java-Module-Project-YP/assets/129396632/b67ba218-930c-4fc2-92c5-b34ce44a24b1">

AleksandrIlinskii commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,43 @@
-        System.out.println("Hello world!");
+
+        int quantityPerson;
+        quantityPerson = 0;
```
Не стоит разделять объявление переменной и ее инициализацию (присваивание значения).
А еще твой 0 нигде не испльзуется, потому что в while ты сразу присваиваешь ей число
<img width="445" alt="image" src="https://github.com/arkadiyashikhmin/Java-Module-Project-YP/assets/129396632/dfd49380-2938-4a55-b1a2-c424559709e1">

AleksandrIlinskii commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,43 @@
+
+            }
+            scanner.nextLine();
+            }
```
Здесь стоит подумать над упрощением вложенности, if else внутри if else

AleksandrIlinskii commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,43 @@
+        sum = Product.totalPrice/quantityPerson;
+
+        System.out.println("Добавленные товары: \n" + Product.nameList);
+        //System.out.println("Количество человек = " + quantityPerson);
```
Закомментированный код мы не пушим

AleksandrIlinskii commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,43 @@
+        System.out.println("Добавленные товары: \n" + Product.nameList);
+        //System.out.println("Количество человек = " + quantityPerson);
+        System.out.println(String.format(messageTemplate,sum) + " рубл" + Rubles.vivod(sum));
+
```
vivod - не лучшее имя для метода. getPlurals подойдет

AleksandrIlinskii commented on src/main/java/Product.java:
```diff
@@ -0,0 +1,43 @@
+    static double price;
+
+    static double totalPrice;
+    static String nameList = "";
```
Выносит переменные стоит только если они используются в нескольких классах / методах. Твой price используется только в этом классе, где один метод

AleksandrIlinskii commented on src/main/java/Product.java:
```diff
@@ -0,0 +1,43 @@
+                    }
+                } else {
+                    System.out.print("Ошибка. Введите цену цифрах ");
+                }
```
If else внутри if else внутри while внутри while 
Тут слишком большая вложенность, нужно или вынести в отдельные функции или подумать о упрощении логики

AleksandrIlinskii commented on src/main/java/Rubles.java:
```diff
@@ -0,0 +1,23 @@
+public class Rubles {
+    static int a; // объявили интовую переменную
```
a - не самое лучшее имя для переменной. Имя должно отражать что там лежит, а что лежит в переменной "а"?

qaashih committed:
Коммит 2
в классе Product в методе calculate://было scanner.nextLine - стало scanner.next()

AleksandrIlinskii commented:
Сломалась логика, теперь он выводит ошибку ввода стоимости до того, как я ее ввел 
<img width="300" alt="image" src="https://github.com/arkadiyashikhmin/Java-Module-Project-YP/assets/129396632/7b263a51-c7f8-47bd-bd9d-2976b851d82a">

AleksandrIlinskii commented:
И теперь он не показывает полное название товара
<img width="421" alt="image" src="https://github.com/arkadiyashikhmin/Java-Module-Project-YP/assets/129396632/c16471d3-545a-49d0-9107-05e50848efec">

qaashih committed:
коммит 3

AleksandrIlinskii commented on src/main/java/Product.java:
```diff
@@ -12,10 +12,18 @@ public class Product {
+                if (name.isEmpty()) { // проверяем на пустое введенное значение
+                    System.out.println("Вы ввели пустое название товара");
+                } else {
+                    //System.out.println("Введено не пустое имя name = " + name + " .Идем дальше");
```
Закоментированный код лучше не пушить
